src/server/pfs/server/driver_v2.go

```diff
@@ -536,3 +581,81 @@ func (d *driverV2) globFileV2(pachClient *client.APIClient, commit *pfs.Commit,
 		return f(fr.Info())
 	})
 }
+
+func (d *driverV2) copyFile(pachClient *client.APIClient, src *pfs.File, dst *pfs.File, overwrite bool) (retErr error) {
+	ctx := pachClient.Ctx()
+	if overwrite {
+		// TODO: after delete merging is sorted out add overwrite support
+		return errors.New("overwrite not yet supported")
+	}
+	srcPrefix := cleanPath(src.Path)
+	dstPrefix := cleanPath(dst.Path)
+	pathTransform := func(x string) string {
+		if !strings.HasPrefix(x, srcPrefix) {
+			panic("cannot apply path transform")
+		}
+		return path.Join(dstPrefix, x[len(srcPrefix):])
+	}
+
+	mr, err := d.storage.NewMergeReader(ctx, []string{commitKey(src.Commit)})
+	if err != nil {
+		return err
+	}
+	return d.withWriter(ctx, dst.Commit.Repo.Name, dst.Commit.ID, func(n int64, dstFs *fileset.Writer) error {
+		return mr.Iterate(func(mfr *fileset.FileMergeReader) error {
+			h, err := mfr.Header()
+			if err != nil {
+				return err
+			}
+			h.Name = cleanPath(h.Name)
+			if !strings.HasPrefix(h.Name, srcPrefix) {
+				return nil
+			}
+			h.Name = pathTransform(h.Name)
+			if err := dstFs.WriteHeader(h); err != nil {
+				return err
+			}
+			return mfr.WriteDataTo(dstFs)
+		})
+	})
+}
+
+func (d *driverV2) inspectFile(pachClient *client.APIClient, file *pfs.File) (fi *pfs.FileInfoV2, retErr error) {
+	commitInfo, err := d.inspectCommit(pachClient, file.Commit, pfs.CommitState_FINISHED)
+	if err != nil {
+		return nil, err
+	}
+	commit := commitInfo.Commit
+	called := false
```

Comment:
I think I mentioned this before, but I would prefer the functionality of this called check to be handled by an exact match functional option on the indexing layer. Also, no globs for inspect file.

Reply:
We would have to rewrite the getTarConditional functionality on top of a MergeReader with that index option. I can make that change as part of the FilterHeader Reader I am working on.